operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+    global_tile_position: Dim<[Ix; 2]>,
+    global_pixel_position: Dim<[Ix; 2]>,
+    tile_size_in_pixels: Dim<[Ix; 2]>,
+}
+
+impl TileInformation {
+    pub fn new(global_size_in_tiles: Dim<[Ix; 2]>, global_tile_position: Dim<[Ix; 2]>, global_pixel_position: Dim<[Ix; 2]>, tile_size_in_pixels: Dim<[Ix; 2]>) -> Self { 
+        Self { global_size_in_tiles, global_tile_position, global_pixel_position, tile_size_in_pixels} 
+    }
+    pub fn global_size_in_tiles(&self) -> Dim<[Ix; 2]> {
+        self.global_size_in_tiles
+    }
+    pub fn global_tile_position(&self) -> Dim<[Ix; 2]> {
+        self.global_tile_position
+    }
+    pub fn global_pixel_position(&self) -> Dim<[Ix; 2]> {
+        self.global_pixel_position
+    }
+    pub fn tile_size_in_pixels(&self) -> Dim<[Ix; 2]> {
+        self.tile_size_in_pixels
+    }
+}
+
+
+
+pub struct GdalSource {
+    time_interval_provider: Vec<TimeInterval>,
+    grid_tile_provider: GdalSourceTileGridProvider,
+    gdal_params: GdalSourceParameters,
+}
+
+impl GdalSource {
+    ///
+    /// Generates a new `GdalSource` from the provided parameters
+    /// TODO: move the time interval and grid tile information generation somewhere else...
+    ///
+    pub fn from_params(params: GdalSourceParameters, time_interval_provider: Vec<TimeInterval>, grid_tile_provider: GdalSourceTileGridProvider) -> Self {
+        GdalSource {
+            time_interval_provider,
+            grid_tile_provider,
+            gdal_params: params
+        }
+    }
+
+    ///
+    /// An iterator which will produce one element per time step and grid tile
+    ///
+    pub fn time_tile_iter(&self) -> impl Iterator<Item=(TimeInterval,TileInformation )> + '_ { // )>
+        let time_interval_iterator = self.time_interval_provider.clone().into_iter();
+        time_interval_iterator.flat_map(
+            move |time| 
+                self.grid_tile_provider.tile_informations().into_iter().map(
+                move |tile| (time, tile)
+            )
+        
+        )
+    }
+
+    ///
+    /// A method to load single tiles from a GDAL dataset.
+    ///
+    pub fn load_tile_data<T: gdal::raster::types::GdalType + Copy>(&self, time_interval: &TimeInterval, tile_information: &TileInformation) -> Result<RasterTile2D<T>> {
+            
+        // snap the time interval start instance to the dataset tick configuration
+        let time_string = self.gdal_params.tick.map(
+            |t| t.snap_datetime(
+                &time_interval.start().as_naive_date_time()
+            ).format(&self.gdal_params.time_format).to_string()
+        );
+        
+        let file_name = self.gdal_params.file_name_with_time_placeholder.replace("%%%_START_TIME_%%%", &time_string.unwrap_or_else(|| "".into()));
+        
+        let path = self.gdal_params.base_path.join(&file_name);
+
+        // open the dataset at path (or 'throw' an error)
+        let dataset = GdalDataset::open(&path).expect("cant open dataset");
```

Comment:
Since you already return a result, why don't you return an `Err(…)` instead of terminating?

Reply:
Yes see above